src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -251,11 +269,25 @@ public Integer invoke(File f, VirtualChannel channel) throws IOException, Interr
         static final StatusCheck STATUS_CHECK_INSTANCE = new StatusCheck();
 
         @Override public Integer exitStatus(FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
+            return exitStatus(workspace, listener);
+        }
+
+        /**
+         * Like {@link #exitStatus(FilePath, Launcher, TaskListener)} but not requesting a {@link Launcher}, which would not be available in {@link #watch} mode anyway.
+         */
+        protected @CheckForNull Integer exitStatus(FilePath workspace, TaskListener listener) throws IOException, InterruptedException {
             FilePath status = getResultFile(workspace);
             return status.act(STATUS_CHECK_INSTANCE);
         }
 
         @Override public byte[] getOutput(FilePath workspace, Launcher launcher) throws IOException, InterruptedException {
+            return getOutput(workspace);
+        }
+
+        /**
+         * Like {@link #getOutput(FilePath, Launcher)} but not requesting a {@link Launcher}, which would not be available in {@link #watch} mode anyway.
+         */
+        protected byte[] getOutput(FilePath workspace) throws IOException, InterruptedException {
```

Comment:
If we're going to rework the way output is done, I'd *really* like to return a stream-able data implementation rather than raw `byte[]` - this ensures we're not allocating giant wads of memory to collect all the data if something produces logs quickly.  Whether it's a Stream, Channel, whatever I do not care as much. 

Consider also something like 'writeOutput' that internally feeds data to a sink.

It's also just plain more elegant and easier to work with in general.

Reply:
Actually... we could still count location that way using a CountingOutputStream I believe.

Reply:
Was already discussed earlier. This method is only used when you specify `returnStdout: true`, a fairly specialized case which is anyway only suited for small amounts of output—in the case of a `workflow-cps` caller, the entire resulting `String` gets saved to `program.dat`.

Reply:
Damnit, we've gone in circles on these PRs so long that I'm repeating myself -- at least I'm consistent though! 😆